public/app/features/dashboard-scene/panel-edit/PanelOptionsPane.tsx

```diff
@@ -76,6 +80,25 @@ export class PanelOptionsPane extends SceneObjectBase<PanelOptionsPaneState> {
       newFieldConfig = restoreCustomOverrideRules(newFieldConfig, cachedFieldConfig);
     }
 
+    if (!this.state.$data) {
+      let ds = getLastUsedDatasourceFromStorage(getDashboardSceneFor(this).state.uid!)?.datasourceUid;
+      if (!ds) {
+        ds = config.defaultDatasource;
+      }
+
+      panel.setState({
+        $data: new SceneDataTransformer({
+          $data: new SceneQueryRunner({
+            datasource: {
+              uid: ds,
+            },
+            queries: [{ refId: 'A' }],
+          }),
+          transformations: [],
+        }),
+      });
+    }
```

Comment:
this feels like the wrong place to fix this bug, the bug is we are rendering the data plane when we should not

Reply:
thinking that we either add something to onChangePanelPlugin to clear data plane when the plugin is a non data plugin.

Reply:
I wanted to add this logic in PanelEditor in [`_updateDataPane`](https://github.com/grafana/grafana/blob/4623a6471bf77036ea3f0f17b0ff4e94272bb01a/public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx#L183). Would it be better to add `$data` and clear `$data` there?